### ISSUE TITLE
PermissionError: [Errno 13] Permission denied: '/run/user/1000'

### ISSUE THREAD

**Issue by MtkN1** [bulletmark/pipxu#8]
## Summary

A `PermissionError` occurs when using the `pipxu install` command.

```console
$ /workspaces/codespaces-blank $ pipx install uv pipxu
...
$ /workspaces/codespaces-blank $ pipxu install hatch
Traceback (most recent call last):
  File "/usr/local/py-utils/bin/pipxu", line 8, in <module>
    sys.exit(main())
  File "/usr/local/py-utils/venvs/pipxu/lib/python3.10/site-packages/pipxu/pipxu.py", line 184, in main
    return args.func(args)
  File "/usr/local/py-utils/venvs/pipxu/lib/python3.10/site-packages/pipxu/commands/install.py", line 60, in main
    with FileLock(lockfile):
  File "/usr/local/py-utils/venvs/pipxu/lib/python3.10/site-packages/filelock/_api.py", line 339, in __enter__
    self.acquire()
  File "/usr/local/py-utils/venvs/pipxu/lib/python3.10/site-packages/filelock/_api.py", line 295, in acquire
    self._acquire()
  File "/usr/local/py-utils/venvs/pipxu/lib/python3.10/site-packages/filelock/_unix.py", line 38, in _acquire
    ensure_directory_exists(self.lock_file)
  File "/usr/local/py-utils/venvs/pipxu/lib/python3.10/site-packages/filelock/_util.py", line 46, in ensure_directory_exists
    Path(filename).parent.mkdir(parents=True, exist_ok=True)
  File "/usr/local/python/3.10.13/lib/python3.10/pathlib.py", line 1175, in mkdir
    self._accessor.mkdir(self, mode)
PermissionError: [Errno 13] Permission denied: '/run/user/1000'
```

## Description

To the best of my knowledge, this error occurs in Unix environments where `systemd` is not running, such as Docker containers and WSL.

The issue seems to arise from the part of the `pipxu install` command that tries to create a lock file.

https://github.com/bulletmark/pipxu/blob/abc02f412d9c9967b55a60c0ef7aab3e3f85c814/pipxu/commands/install.py#L56-L60

In Unix environments, the path configured by `platformdirs.user_runtime_path()` becomes `/run/user/xxxx`. However, in the targeted environments, `/run/user/xxxx` does not exist. Moreover, the `/run/user` directory is not accessible to normal users.

As a result, users are forced to run `sudo mkdir /run/user/xxxx && sudo chown -R $USER:$USER /run/user/xxxx` before using `pipxu install`. This is not an ideal solution.

The easiest way to reproduce this issue is to launch a **Blank** GitHub Codespace: https://github.com/codespaces

## Solution

How about generating the lock file under `$USER/.local`? From my understanding, either of the following paths would be suitable:

```py
>>> platformdirs.user_data_path()
PosixPath('/home/codespace/.local/share')
>>> platformdirs.user_state_path()
PosixPath('/home/codespace/.local/state')
```

I would be happy to create a pull request once we reach an agreement on this change. Thank you for the great work on `pipxu`!

**Comment by bulletmark** [comment]
Thanks for your thorough report about this!

Actually, as I state in commit 84ffda4, this was really a bug. I have released version 1.18 with this fix.

**Comment by MtkN1** [comment]
Thanks for the speedy fix 🚀  Keep up the great work!
